### PR TITLE
Blog App: Processing data

### PR DESCRIPTION
**Pull request by manzitresor**
### Implemented Feature

- [ ] Users

  - A method that returns the 3 most recent posts for a given user.

- [ ] Posts

  - A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.

- [ ] Comments

  - A method that updates the comments counter for a post.

- [ ] Likes

  - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,12 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+  def posts_counter
+    author.increment!(:postsCounter)
+  end
+
```
Comment (mukhammadrizooff) [inline app/models/post.rb:13]: - [ ] [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private 👍🏼

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
 class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (mukhammadrizooff) [inline app/models/user.rb]: - [ ] [REQUIRED] Good job so far! The associations between models should be set up correctly according, please implement `foreign key` correctly as below 🙏🏼

```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
   belongs_to :user
   belongs_to :post
```
Comment (mukhammadrizooff) [inline app/models/like.rb]: - [ ] [REQUIRED] Good job so far! The associations between models should be set up correctly according, please implement `:author` correctly as below 🙏🏼

```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,4 +1,10 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
```
Comment (mukhammadrizooff) [inline app/models/comment.rb]: - [ ] [REQUIRED] Good job so far! The associations between models should be set up correctly according, please implement `:author` correctly as below 🙏🏼

```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @manzitresor, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by manzitresor** [comment]
HI @mukhammadrizooff  
thank you for review feedback.

**Review by danmainah — APPROVED** [review]
Hi @manzitresor , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the changes requested to ensure all associations are setup correctly.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**